plugins/aws/cli_provisioner.go

```diff
@@ -0,0 +1,58 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/1Password/shell-plugins/sdk"
+)
+
+const defaultProfileName = "default"
+
+type awsCLIProvisioner struct {
+}
+
+func NewAwsCLIProvisioner() sdk.Provisioner {
+	return awsCLIProvisioner{}
+}
+
+func (p awsCLIProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	profileName := defaultProfileName
+
+	if profile := stripAndReturnProfileFlag(out); profile != nil {
+		profileName = *profile
+	} else if profile := os.Getenv("AWS_PROFILE"); profile != "" {
+		profileName = profile
+	}
+	NewSTSProvisioner(profileName).Provision(ctx, in, out)
+}
+
+func stripAndReturnProfileFlag(out *sdk.ProvisionOutput) *string {
+	for i, arg := range out.CommandLine {
+		if arg == "--profile" {
```

Comment:
I am not sure if the AWS CLI uses this anywhere, but if `--profile` is used after `--`, it should not be interpreted as a flag by the AWS CLI. Just like how in `op run -- aws --profile <profile>` `--profile` is not interpreted as a flag for `op`, but as an argument instead.

Reply:
Makes sense, handled that case now!